### PR TITLE
Added a component

### PR DESCRIPTION
**Pull request by simonachkar**

### CODE REVIEW

File: src/components/counter.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+function Counter() {
+  // Declare a state variable called 'count' and a function to update it called 'setCount'
+  const [count, setCount] = useState(0);
+
+  const increment = () => {
+    setCount(count + 1);
+  };
+
+  const decrement = () => {
```
Comment (simonachkar) [inline src/components/counter.js:11]: Consider adding error handling for potential integer overflow/underflow. While unlikely with simple increment/decrement, it's good practice to be aware of potential limits, especially if the component might be extended with larger increments/decrements in the future. A check to ensure `count` remains within safe integer bounds could be added.

Example:
```javascript
const increment = () => {
  if (count < Number.MAX_SAFE_INTEGER) {
    setCount(count + 1);
  }
};

const decrement = () => {
  if (count > Number.MIN_SAFE_INTEGER) {
    setCount(count - 1);
  }
};
```

File: src/components/counter.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    setCount(count - 1);
+  };
+
+  return (
+    <div>
+      <h2>Simple Counter</h2>
+      <p>Count: {count}</p>
+      <button onClick={increment}>Increment</button>
+      <button onClick={decrement}>Decrement</button>
```
Comment (simonachkar) [inline src/components/counter.js:20]: The UI/UX can be improved. The buttons currently have no visual feedback when clicked. Consider adding visual styling (e.g., using CSS classes) to the buttons for hover and active states to provide a better user experience. Also, consider adding labels to the buttons with `aria-label` attributes for accessibility.

Example:
```html
<button aria-label="Increment counter" onClick={increment}>Increment</button>
<button aria-label="Decrement counter" onClick={decrement}>Decrement</button>
```

File: src/components/counter.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+export default Counter;
+
+/**
+ * Here are some related topics you might find interesting:
+ * - useState Hook -> https://react.dev/reference/react/useState
+ * - Functional Components -> https://react.dev/learn/your-first-component#functional-components
+ * - Event Handling in React -> https://react.dev/learn/responding-to-events
+ * - JSX (JavaScript XML) -> https://react.dev/learn/writing-markup-with-jsx
+ * - Props (Properties) -> https://react.dev/learn/passing-props-to-a-component
+ */
```
Comment (simonachkar) [inline src/components/counter.js:34]: The comments block at the end of the file, while helpful for learning, doesn't provide any context within the code itself. While useful, it would be better to remove these or to provide them separately or in a documentation file. They clutter up the component code without providing immediate value or explanation for the current code.

Consider removing them, or move them to a README file for the component, or a separate documentation page if you plan to use the component in a larger context.

If you wanted to keep them, an alternative is to move them to the description of the component when it is being imported/used in another component, like so:

```javascript
/**
 * Counter component.
 *
 * Here are some related topics you might find interesting:
 * - useState Hook -> https://react.dev/reference/react/useState
 * - Functional Components -> https://react.dev/learn/your-first-component#functional-components
 * - Event Handling in React -> https://react.dev/learn/responding-to-events
 * - JSX (JavaScript XML) -> https://react.dev/learn/writing-markup-with-jsx
 * - Props (Properties) -> https://react.dev/learn/passing-props-to-a-component
 */
import Counter from './Counter';
```

File: src/components/counter.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const increment = () => {
+    setCount(count + 1);
+  };
+
+  const decrement = () => {
+    setCount(count - 1);
+  };
+
+  return (
+    <div>
```
Comment (simonachkar) [inline src/components/counter.js:16]: Consider adding unit tests to verify the functionality of the `increment` and `decrement` functions. This will help ensure the component behaves as expected and prevent regressions in the future.  Tests should cover both positive and negative increments/decrements as well as edge cases.  A testing framework like Jest and React Testing Library are popular choices.

Example:
```javascript
// Counter.test.js
import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './Counter';

describe('Counter Component', () => {
  test('increments count when increment button is clicked', () => {
    render(<Counter />);
    const incrementButton = screen.getByText('Increment');
    fireEvent.click(incrementButton);
    const countElement = screen.getByText('Count: 1');
    expect(countElement).toBeInTheDocument();
  });

  test('decrements count when decrement button is clicked', () => {
    render(<Counter />);
    const decrementButton = screen.getByText('Decrement');
    fireEvent.click(decrementButton);
    const countElement = screen.getByText('Count: -1');
    expect(countElement).toBeInTheDocument();
  });
});
```

**Review by simonachkar — COMMENTED** [review]
This PR introduces a new `Counter` component, which is a basic implementation of a counter with increment and decrement buttons. The code appears functional and uses React's `useState` hook correctly. However, there are some areas that could be improved for better maintainability and user experience.
